src/scripts/main.js

```diff
@@ -1,3 +1,193 @@
 'use strict';
 
-// write your code here
+const table = document.querySelector('table');
+let score = 0;
+
+let board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+const rows = 4;
+const columns = 4;
+
+const button = document.querySelector('button');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const messageStart = document.querySelector('.message-start');
+
+button.addEventListener('click', () => {
+  if (button.classList.contains('start')) {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.innerText = 'Restart';
+
+    messageStart.classList.add('hidden');
+    setGame();
+    setGame();
+    updateGame();
+  } else {
+    if (button.classList.contains('restart')) {
+      button.classList.remove('restart');
+      button.classList.add('start');
+      messageStart.classList.remove('hidden');
+      button.innerText = 'Start';
+
+      board = [
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ];
+
+      messageLose.classList.add('hidden');
+      messageWin.classList.add('hidden');
+      score = 0;
+
+      updateGame();
```

Comment:
You call  `updateGame()` in both cases. So just call it at the end of the function once